# Ok Cupid Dataset EDA

This dataset was picked on the name alone to do an initial clean and EDA.
The dataset is completly unfamiliar to me - I want to practice familiarising myself with the set and seeing what I can learn about it, to see what further investigation can stem from it. 

In [6]:
# initial imports and check file is readable
import pandas as pd
okcupid = pd.read_csv('OkCupid_kaggle_dataset.zip')


In [7]:
print(okcupid.head((10)))

   age     status sex orientation       body_type               diet  \
0   22     single   m    straight  a little extra  strictly anything   
1   35     single   m    straight         average       mostly other   
2   38  available   m    straight            thin           anything   
3   23     single   m    straight            thin         vegetarian   
4   29     single   m    straight        athletic                NaN   
5   29     single   m    straight         average    mostly anything   
6   32     single   f    straight             fit  strictly anything   
7   31     single   f    straight         average    mostly anything   
8   24     single   f    straight             NaN  strictly anything   
9   37     single   m    straight        athletic    mostly anything   

       drinks      drugs                          education  \
0    socially      never      working on college/university   
1       often  sometimes              working on space camp   
2    socially     

In [2]:
okcupid.shape

(59946, 31)

This set has 59946 rows and 31 columns

## Duplicates check

Index(['age', 'status', 'sex', 'orientation', 'body_type', 'diet', 'drinks',
       'drugs', 'education', 'ethnicity', 'height', 'income', 'job',
       'last_online', 'location', 'offspring', 'pets', 'religion', 'sign',
       'smokes', 'speaks', 'essay0', 'essay1', 'essay2', 'essay3', 'essay4',
       'essay5', 'essay6', 'essay7', 'essay8', 'essay9'],
      dtype='object')


age  status    sex orientation body_type             diet  \
count   59946.000000   59946  59946       59946     54650            35551   
unique           NaN       5      2           3        12               18   
top              NaN  single      m    straight   average  mostly anything   
freq             NaN   55697  35829       51606     14652            16585   
mean       32.340290     NaN    NaN         NaN       NaN              NaN   
std         9.452779     NaN    NaN         NaN       NaN              NaN   
min        18.000000     NaN    NaN         NaN       NaN              NaN   
25%        26.000000     NaN    NaN         NaN       NaN              NaN   
50%        30.000000     NaN    NaN         NaN       NaN              NaN   
75%        37.000000     NaN    NaN         NaN       NaN              NaN   
max       110.000000     NaN    NaN         NaN       NaN              NaN   

          drinks  drugs                          education ethnicity  ...  \
count      56961  45866                              53318     54266  ...   
unique         6      3                                 32       217  ...   
top     socially  never  graduated from college/university     white  ...   
freq       41780  37724                              23959     32831  ...   
mean         NaN    NaN                                NaN       NaN  ...   
std          NaN    NaN                                NaN       NaN  ...   
min          NaN    NaN                                NaN       NaN  ...   
25%          NaN    NaN                                NaN       NaN  ...   
50%          NaN    NaN                                NaN       NaN  ...   
75%          NaN    NaN                                NaN       NaN  ...   
max          NaN    NaN                                NaN       NaN  ...   

        essay0        essay1     essay2    essay3  essay4  essay5     essay6  \
count    54458         52374      50308     48470   49409   49096      46175   
unique   54347         51516      48625     43520   49257   48961      43583   
top          .  enjoying it.  listening  my smile  ask me  family  my future   
freq        12            61         82       529      16       6        161   
mean       NaN           NaN        NaN       NaN     NaN     NaN        NaN   
std        NaN           NaN        NaN       NaN     NaN     NaN        NaN   
min        NaN           NaN        NaN       NaN     NaN     NaN        NaN   
25%        NaN           NaN        NaN       NaN     NaN     NaN        NaN   
50%        NaN           NaN        NaN       NaN     NaN     NaN        NaN   
75%        NaN           NaN        NaN       NaN     NaN     NaN        NaN   
max        NaN           NaN        NaN       NaN     NaN     NaN        NaN   

                  essay7  essay8        essay9  
count              47495   40721         47343  
unique             45548   39323         45440  
top     out with friends  ask me  you want to.  
freq                  89      45           200  
mean                 NaN     NaN           NaN  
std                  NaN     NaN           NaN  
min                  NaN     NaN           NaN  
25%                  NaN     NaN           NaN  
50%                  NaN     NaN           NaN  
75%                  NaN     NaN           NaN  
max                  NaN     NaN           NaN  

[11 rows x 31 columns]